src/components/Application/Navigation.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { Button, Segment, Sticky } from 'semantic-ui-react'
+import {
+  MethodRevalidate,
+  MethodToCallProps,
+  SectionAndPage,
+  SectionDetails,
+  SectionsStructureNEW,
+} from '../../utils/types'
+import strings from '../../utils/constants'
+import { useRouter } from '../../utils/hooks/useRouter'
+
+interface NavigationProps {
+  current: SectionAndPage
+  sections: SectionsStructureNEW
+  serialNumber: string
+  requestRevalidation: MethodRevalidate
+}
+
+const Navigation: React.FC<NavigationProps> = ({
+  current,
+  sections,
+  serialNumber,
+  requestRevalidation,
+}) => {
+  const { push } = useRouter()
+
+  const currentSectionDetails = sections[current.sectionCode].details
+
+  const hasNextSection = Object.values(sections).find(
+    ({ details: { index } }) => index > currentSectionDetails.index
+  )
+  const hasPreviousSection = Object.values(sections).find(
+    ({ details: { index } }) => index < currentSectionDetails.index
+  )
+
+  const isFirstPage = current.pageNumber - 1 === 0 && !hasPreviousSection
+  const isLastPage = current.pageNumber + 1 > currentSectionDetails.totalPages && !hasNextSection
+
+  const getPreviousSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber > 1) return { sectionCode, pageNumber: pageNumber - 1 }
+    else {
+      const previousSectionDetails = Object.values(sections).reduce(
```

Comment:
as per earlier comment hasPreviousSection can be renamed to previousSection and used here

Reply:
Done. Thanks : )